### DISCUSSION TITLE
NOW() and TIMESTAMP() commands are not accurate

### DISCUSSION THREAD

**Discussion by alexpetralia** [dbeaver/dbeaver#364] [Q&A] [answered]
Hi,

If I run `SELECT NOW()` or `SELECT CURRENT_TIMESTAMP` (postgres), the result window does not update the time. It remains frozen. However, results are fetched and the ms time changes, so it appears to be querying something. At some point later, running that exact query again returns an updated time, though subsequently running it will show it is frozen once again. Running the same queries in `pgadmin` update as you'd expect - the queries are actually hitting the server. Does DBeaver overwrite these timestamp queries to use the local machine's information instead of directly submitting the queries to the server?

![image](https://cloud.githubusercontent.com/assets/9990778/14507652/9ed534dc-0191-11e6-855c-d95e6375ecfd.png)

![image](https://cloud.githubusercontent.com/assets/9990778/14507656/a94bafc2-0191-11e6-9d19-e17ce80bee4e.png)

Thanks.

**Comment by serge-rider** [comment, marked as answer]
This is not a DBeaver issue. It is how PostgreSQL works.
now() or CURRENT_TIMESTAMP is the same as transaction_timestamp().
Time will change once you will commit or rollback active transaction.

pgadmin works in autocommit state by default so time differs on each execution.
You can switch to autocommit state in DBeaver or use clock_timestamp() function.

http://stackoverflow.com/questions/3363376/how-to-get-a-real-time-within-postgresql-transaction

**Comment by alexpetralia** [comment]
Oddly, my connection is set to autocommit (which I confirmed with `SHOW autocommit`) and the issue is still there. However, that is a committing issue and doesn't represent a problem with the program. `CLOCK_TIMESTAMP()` solved my issue, thank you!
